### PR TITLE
Add support to http headers

### PR DESCRIPTION
**Pull request by charafau**
- Updated project to support newer quiver
- Removed deprecated analysis options
- Added file extension to .analysis_options
- Added support for newer dart
- Added headers to constructor

### CODE REVIEW

**Comment by Hixie** [comment]
@kf6gpe were you planning on moving this repo to the packages repo?

File: lib/network.dart [hunk 1]
```diff
@@ -115,8 +117,12 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
       io.HttpClientRequest request;
       try {
         request = await _client.getUrl(instructions.uri).timeout(instructions.timeout);
+        if(headers != null){
```
Comment (Hixie) [inline lib/network.dart]: nit: we put spaces around the expression, as in `if (headers != null) {`

**Comment by Hixie** [comment]
LGTM

**Comment by vasilich6107** [comment]
This PR waits to be resolved for almost two years....

**Comment by stuartmorgan** [comment]
Looks like this was never updated for the review nit. It now has a number of conflicts; @charafau Is this something you're interested in updating (ideally after https://github.com/flutter/flutter_image/pull/22 lands) and getting landed?

**Comment by charafau** [comment]
@stuartmorgan  I think it was already done since it got LGTM, I will update the code today and upload it so it can be merged

**Comment by stuartmorgan** [comment]
> @stuartmorgan I think it was already done since it got LGTM, I will update the code when above PR gets merged to null safety (or create new one)

But it was LGTM'd with the understanding that the formatting issue that was raised would be addressed, so it's not actually done until that's addressed.

**Comment by charafau** [comment]
That's my bad. I will fix it again then null safety PR lands 👍

**Comment by stuartmorgan** [comment]
I've migrated `flutter_image` to the flutter/packages repository, so we're archiving this repo; you'll want to re-submit this a PR there once you have time to update it for null safety. Sorry for all the churn here!

**Pull request closed without merging**